tools/benchcomp/test/regression/pass/2_variant_run/Makefile

```diff
@@ -0,0 +1,8 @@
+# Copyright Kani Contributors
```

Comment:
Just in case it wasn't clear: the top-level script that runs both the unit and regression tests is also a Makefile (but in disguise, it's a script called `test/run` but it has `#!/usr/bin/make -f` as shebang)

For both the top-level script and each of the regression tests: I wanted a way of doing regression testing with few dependencies and that anybody can add to easily. Shell scripts or Makefiles seem like the right level here, python is really heavyweight when all you want to do is run some commands.

I decided against shell scripts because every command here could possibly fail, both `benchcomp` itself and the `yq` commands being used as assertions. In shell, we would either need to check `$?` after every command (cumbersome, again I want it to be easy for all of us to add to this), or I could use `set -e` or `TRAP err` with all of their downsides (like they don't actually fail from inside a loop/pipe/etc). So a Makefile seemed like a natural fit.

Happy to change to something else if you have strong opinions against, but for regression tests I'd hope for something where we can write a command and some assertions with little boilerplate and with little chance of forgetting to do the actual error checking/bailing out.

Reply:
I think most people are not familiar with Makefiles. I would prefer if we use shell, python or rust. We usually use `set -e`. You don't seem to be using a loop or a pipe here, so I don't see why that wouldn't work.

Reply:
you could also set `set -o pipefail` if you are worried about pipe masking the error.

Reply:
OK. I'll convert both these to shell scripts.

One other advantage of makefiles I forgot to mention: they offer a very natural way of expressing concurrency, by describing what tasks depend on what other tasks. So you can naturally parallelize the unit tests without external dependencies.

With a shell script what I'll try to do is test to see if either `xargs` or GNU `parallel` is installed, and pipe a list of unit test commands to them if so, or pipe into a while loop if not.

Reply:
Or actually maybe I'll just write a simple loop for now to get this merged and add parallelism later once we have more than two tests :P

Reply:
I get your point. Makefiles have benefits, but I don't see you using any of them in this script.

Maybe you have a use case in mind, but please try to keep it simple. Only add complexity if the benefit is worth it, because makefiles will make things harder to maintain.

Reply:
I've replaced the testing suite. The regression test is now also using python's stdlib `unittest` module (although it's really invoking the binary, not running unit tests). I've added three regression tests, and the top-level testing script is now a trivial shellscript that runs invokes python's unit-testing toplevel.